perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_elixirBonaFideStatus.java

```diff
@@ -0,0 +1,100 @@
+ */
+public class urn_perun_user_attribute_def_virt_elixirBonaFideStatus extends UserVirtualAttributesModuleAbstract implements UserVirtualAttributesModuleImplApi {
+
+	private final static Logger log = LoggerFactory.getLogger(urn_perun_user_attribute_def_virt_elixirBonaFideStatus.class);
+
+	private final Pattern addUserExtSource = Pattern.compile("UserExtSource:\\[(.*)] added to User:\\[(.*)]");
+	private final Pattern removeUserExtSource = Pattern.compile("UserExtSource:\\[(.*)] removed from User:\\[(.*)]");
+	private final Pattern extSourceRems = Pattern.compile("cz.metacentrum.perun.core.impl.ExtSourceREMS");
+	private final Pattern affiliationsAttribute = Pattern.compile("Attribute:\\[(.*)friendlyName=<eduPersonScopedAffiliations>(.*)] set for User:\\[(.*)]");
+
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDefinition attributeDefinition) throws InternalErrorException {
+		Attribute attribute = new Attribute(attributeDefinition);
+		String url = "http://www.ga4gh.org/beacon/bonafide/ver1.0";
+
+		try {
+			Attribute statusAttr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_DEF + ":elixirBonaFideStatusREMS");
+			String attrDefValue = (String) statusAttr.getValue();
+			if (attrDefValue != null && !attrDefValue.isEmpty()) {
+				attribute.setValue(attrDefValue);
+			} else {
+				Attribute uesAttribute = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_VIRT + ":eduPersonScopedAffiliations");
+				if (uesAttribute.getValue() != null) {
+					List<String> affiliations = (List<String>) uesAttribute.getValue();
+					for (String affiliation : affiliations) {
+						if (affiliation != null && affiliation.startsWith("faculty@")) {
+							attribute.setValue(url);
+							break;
+						}
+					}
+				}
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Cannot read elixirBonaFideStatusREMS from user " + user.getId(), e);
+		}
+		return attribute;
+	}
+
+	@Override
+	public List<String> resolveVirtualAttributeValueChange(PerunSessionImpl sess, String message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
+		List<String> resolvingMessages = new ArrayList<>();
+		if (message == null) return resolvingMessages;
+
+		Matcher addUserExtSourceMatcher = addUserExtSource.matcher(message);
+		Matcher removeUserExtSourceMatcher = removeUserExtSource.matcher(message);
+		Matcher extSourceRemsMatcher = extSourceRems.matcher(message);
+		Matcher affiliationsAttributeMatcher = affiliationsAttribute.matcher(message);
+
+		User user;
+		Attribute attrVirtRemsLogins;
+
+		if (affiliationsAttributeMatcher.find() || (
+				extSourceRemsMatcher.find() && (addUserExtSourceMatcher.find() || removeUserExtSourceMatcher.find()))) {
```

Comment:
So if I understand it right there are these possible ways:
- attribute with friendly name "..Affiliations" was set for "user
- extSourceREMS was add to user
- extSourceREMS was remove for user

There is still missing part with removing value from affiliations attribute I think.